ppl/zqd/recruiter/recruiter.go

```diff
@@ -0,0 +1,191 @@
+package recruiter
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"sync"
+	"time"
+)
+
+// WorkerPool is the internal state of the recruiter system
+// The pools are exported only for use in unit tests
+type WorkerPool struct {
+	mu           sync.Mutex                // one lock for all three maps
+	freePool     map[string]WorkerDetail   // Map of all free workers
+	nodePool     map[string][]WorkerDetail // Map of nodes of slices of free workers
+	reservedPool map[string]WorkerDetail   // Map of busy workers
+	SkipSpread   bool                      // option to test algorithm performance
+}
+
+type WorkerDetail struct {
+	Addr     string
+	NodeName string
+}
+
+func NewWorkerPool() *WorkerPool {
+	rand.Seed(time.Now().UnixNano()) // for shuffling nodePool keys
```

Comment:
Other package can undo this. Avoid that by creating a math.Rand for this package.